### PR TITLE
Upgrade airtap and use GitHub Actions

### PR DESCRIPTION
**Pull request by vweevers**
- Upgrade airtap to v4 (see its [upgrade guide](https://github.com/airtap/airtap/blob/master/UPGRADING.md))
- Move Sauce Labs tests from Travis to GitHub Actions
- Use playwright for headless local testing

Complements #418.

### CODE REVIEW

File: .github/workflows/sauce.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        env:
+          # Download Sauce Connect binary now instead of on first run
+          SAUCE_CONNECT_DOWNLOAD_ON_INSTALL: true
+      - name: Add host
+        run: echo "127.0.0.1 airtap.local" | sudo tee -a /etc/hosts
+      - name: Test
+        run: npm run test-browsers
+        env:
+          SAUCE_USERNAME: ${{ secrets.SAUCE_USERNAME }}
+          SAUCE_ACCESS_KEY: ${{ secrets.SAUCE_ACCESS_KEY }}
```
Comment (vweevers) [inline .github/workflows/sauce.yml:24]: @mcollina Can you add these two secrets?

**Comment by vweevers** [comment]
Travis fails on npm install because playwright (which has an install script) requires node 10. I'll remove playwright for now, unless someone has an idea for a workaround.

**Comment by mcollina** [comment]
They keys are set up

**Comment by vweevers** [comment]
Build failed:

> Error: 23 Sep 19:50:19 - Sauce Connect could not establish a connection.

Will see if I can replicate that on another repo that has an almost identical setup, and no sauce connect problems.

**Comment by vweevers** [comment]
With debug output enabled, the problem [seems to be the credentials](https://github.com/nodejs/readable-stream/runs/1157278209?check_suite_focus=true#step:6:148):

```
ERROR: connecting to https://saucelabs.com/rest/v1/***/tunnels: HTTP response code indicated failure, reply: {"message": "Not authorized"}.
ERROR: can't reach https://saucelabs.com/rest/v1/***/tunnels, please check your username/access key, firewall and proxy settings.
```

@mcollina Could you please double-check the credentials? You can also try it locally:

```
SAUCE_USERNAME=xx SAUCE_ACCESS_KEY=xx DEBUG=airtap* npm run test-browsers
```

**Comment by mcollina** [comment]
Thanks! Fixed and now the tests are passing!

**Comment by mcollina** [comment]
Running `npm test-browser` I'm getting:

```
Hostname 'airtap.local' must resolve to 127.0.0.1. Please add an entry to your hosts file.
```

**Comment by vweevers** [comment]
@mcollina That hostname error is expected. Some browsers don't route localhost traffic through the Sauce Connect tunnel. To work around that, airtap uses a custom hostname that defaults to `airtap.local` and should resolve to 127.0.0.1. Only needed for sauce, not for playwright and others.

The debug steps I added in the last commit should be removed, especially `./node_modules/sauce-connect-launcher/sc/sc-4.6.2-linux/bin/sc` which will break when a new Sauce Connect version is released by Sauce Labs.

**Comment by mcollina** [comment]
oh gosh... can you send a fresh PR?

**Pull request merged**